src/platform/Linux/BLEManagerImpl.cpp

```diff
@@ -804,15 +804,24 @@ void BLEManagerImpl::OnDeviceScanned(BluezDevice1 * device, const chip::Ble::Chi
 
 void BLEManagerImpl::OnScanComplete()
 {
-    if (mBLEScanConfig.mBleScanState != BleScanState::kScanForDiscriminator &&
-        mBLEScanConfig.mBleScanState != BleScanState::kScanForAddress)
+    switch (mBLEScanConfig.mBleScanState)
     {
+    case BleScanState::kNotScanning:
         ChipLogProgress(Ble, "Scan complete notification without an active scan.");
-        return;
+        break;
+    case BleScanState::kScanForAddress:
+    case BleScanState::kScanForDiscriminator:
+        mBLEScanConfig.mBleScanState = BleScanState::kNotScanning;
+        ChipLogProgress(Ble, "Scan complete. No matching device found.");
+        break;
+    case BleScanState::kConnecting:
```

Comment:
is it normal to have a scan complete when in connecting state?

Should _the "scan complete notification without an active scan" be Progress or should we make it error since it is some internal unexpected error?

Reply:
This is normal to scan complete when in the connecting state. In `void BLEManagerImpl::OnDeviceScanned`, there is a call for `StopScan`, which later calls `OnScanComplete`. Before it, the state is changed to kConnecting in  [OnDeviceScanned](https://github.com/project-chip/connectedhomeip/blob/8fafbab9a57507b6854a08a62ec38f55c2ef6942/src/platform/Linux/BLEManagerImpl.cpp#L798) function

The "scan complete notification without an active scan"  was the expected state.  An error occurred later, so I wonder why I should make an error.